src/hooks/useMaciSignup.ts

```diff
@@ -44,12 +52,23 @@ export const useMaciSignup = (): IUseMaciSignUpData => {
       return;
     }
 
-    const { stateIndex: index, hash } = await signup({
-      maciPubKey: data.publicKey,
-      maciAddress: config.maciAddress!,
-      sgDataArg: attestationId,
-      signer: signer as unknown as Signer,
-    });
+    setIsLoading(true);
+
+    let index, hash;
```

Comment:
You still have unneeded mutations here and options.onError is not added to useCallback deps.

Reply:
Not sure what do you mean by "unneeded mutations", is that refer to the `setIsLoading`? I just want to prevent user from clicking the button (since it will display `signup` during the signup transaction, which might be confusing)

Reply:
You don't need to use `let` here. It's just reassigning variables in this case, so it has no sense because you can set this data inside try.